### ISSUE TITLE
Messed up my fish shell... (built in command mistake)

### ISSUE THREAD

**Issue by OakNinja** [fish-shell/fish-shell#6953]
Ok, this is really embarassing - I was fiddling around with using built in commands in a function, and really messed up big time. 

`set -q MAKE_ME_FISH_PARSER; or set MAKE_ME_FISH_PARSER "fish_default"`
`if string match -q "fish_default" $MAKE_ME_FISH_PARSER`
     `set targets (__fish_print_make_targets "" $filename)`

Not being too familiar with writing fish functions yet, I realized I released hell in some way. 

`functions --all | grep make` 
outputs 
`__fish_make_completion_signals`
and the shell is not working as intended. 

This is the full commit that caused the mess: https://github.com/OakNinja/MakeMeFish/commit/4565c59cd70acb46caa2444071d7c9d046843eb2

Using iTerm 3.3.9, Fish 3.1.1, on OS X. 

I've tried `sh -c 'env HOME=$(mktemp -d) fish'` with the same result. I've deleted my ~/.config/fish/ directory and ran 'brew uninstall fish' 'brew reinstall fish' as well as looking through environment variables. I don't know how to get everything back to normal. 

I've probably overlooked something obvious and fundamental...

**Comment by zanchey** [comment]
> the shell is not working as intended

In what sense? I can't see anything obvious on first inspection.

**Comment by OakNinja** [comment]
https://github.com/fish-shell/fish-shell/blob/master/share/completions/make.fish contains two functions, __fish_print_make_targets and __fish_make_completion_signals. 

After I pulled and ran the code above, the function __fish_print_make_targets is no longer present and completions does not work, short cuts does not work, and the shell is unresponsive.

 ~> __fish_print_make_targets
fish: Unknown command: __fish_print_make_targets

**Comment by krobelus** [comment]
Those functions are only loaded as the completions for `make` are. If you type `make ` and Tab __fish_print_make_targets will be defined.

You could do `source $__fish_datadir/completions/make.fish` in your script, or simply copy the functions you need (and possibly rename them for good measure).

**Comment by OakNinja** [comment]
> Those functions are only loaded as the completions for `make` are. If you type `make ` and Tab __fish_print_make_targets will be defined.
> 
> You could do `source $__fish_datadir/completions/make.fish` in your script, or simply copy the functions you need (and possibly rename them for good measure).

Thanks - I realized this this morning. 

Long story short - when I installed my package on another machine, the __fish_print_make_targets function could not be found. Not realizing that make.fish needed to be sourced in the current session, I assumed I did something wierd by mistake. Trying to fix my problem by resetting the environment, and as a result, I installed fish 3.1.1. This in turn broke 'jethrokuan/fzf' which made me believe I really messed something fundamental up. Hours later, downgraded to fish 3.1.0, and now everything works as expected again. 

I would really appreciate fish being versioned at least one minor and one major version back in homebrew, eg. enable for `brew install fish@3.1.0` to work.
